src/components/molecules/Assignment/Assignment.tsx

```diff
@@ -167,9 +196,74 @@ const Assignment: FC<AssignmentProps> = ({
     }
   }, [t])
 
-  // TODO: shouldShowStartTimeFields no info about where to take this from yet
-  const shouldShowStartTimeFields =
-    project?.type_classifier_value?.project_type_config?.is_start_date_supported
+  const handleUpdateAssignment = useCallback(
+    async (payload: AssignmentPayload) => {
+      try {
+        await updateAssignment(payload)
+        showNotification({
+          type: NotificationTypes.Success,
+          title: t('notification.announcement'),
+          content: t('success.assignment_updated'),
+        })
+      } catch (errorData) {
+        showValidationErrorMessage(errorData)
+      }
+    },
+    [t, updateAssignment]
+  )
+
+  const textAreaElement =
+    document?.getElementById(`textarea_${id}`) || document.body
+
+  const formattedDeadline = useMemo(
+    () => getBEDate(deadline_at || projectDeadline),
+    [deadline_at, projectDeadline]
+  )
+
+  const handleAddComment = useCallback(
+    (event: Event) => {
+      const comment = getValues('comments')
+      const isCommentChanged = !isEqual(comment, comments)
+
+      if (textAreaElement.contains(event.target as Node) && isCommentChanged) {
+        handleUpdateAssignment({
+          deadline_at: formattedDeadline,
+          comments: getValues('comments'),
+        })
+      }
+    },
+
+    [
+      comments,
+      formattedDeadline,
+      getValues,
+      handleUpdateAssignment,
+      textAreaElement,
+    ]
+  )
+  const handleAddDateTime = useCallback(
+    (value: { date: string; time: string }) => {
+      const { date, time } = value
+      const dateTime = dayjs.utc(`${date} ${time}`, 'DD/MM/YYYY HH:mm')
+      const formattedDateTime = dateTime.format('YYYY-MM-DDTHH:mm:ss[Z]')
+      const isDeadLineChanged = !isEqual(formattedDeadline, formattedDateTime)
+
+      if (isDeadLineChanged) {
+        handleUpdateAssignment({
+          deadline_at: formattedDateTime,
+        })
+      }
+    },
+
+    [formattedDeadline, handleUpdateAssignment]
+  )
+
+  useEffect(() => {
```

Comment:
BLOCKER
There is a problem here. During the first mounting of the component, textAreaElement points to the <body>, and therefore does not update when used directly after opening the page. Only after a state re-render (for example, by changing the deadline) does it get updated to the correct text input element.